### PR TITLE
pytest: introduce runTests function

### PR DESCRIPTION
**Pull request by FRidh**
###### Motivation for this change

A simple function that generates a `checkPhase`. It includes a
`disabledTests` parameter to conveniently disable tests.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by grahamc** [comment]
@GrahamcOfBorg eval

**Comment by grahamc** [comment]
Ofborg's new maintainer ping algorithm would have requested a review from:

 - @nixy for pkgs.python27Packages.astor, pkgs.python37Packages.astor, pkgs.python27Packages.astor, pkgs.python37Packages.astor, pkgs.python27Packages.astor, pkgs.python27Packages.astor, pkgs.python37Packages.astor, pkgs.python37Packages.astor
 - @knedlsepp for pkgs.python37Packages.pandas, pkgs.python27Packages.pandas, pkgs.python27Packages.pandas, pkgs.python37Packages.pandas, pkgs.python37Packages.pandas, pkgs.python37Packages.pandas, pkgs.python27Packages.pandas, pkgs.python27Packages.pandas
- @fridh for pkgs.python37Packages.datashape, pkgs.python37Packages.datashape, pkgs.python37Packages.pandas, pkgs.python27Packages.traitlets, pkgs.python37Packages.traitlets, pkgs.python27Packages.pandas, pkgs.python37Packages.traitlets, pkgs.python37Packages.traitlets, pkgs.python27Packages.traitlets, pkgs.python27Packages.pandas, pkgs.python27Packages.datashape, pkgs.python27Packages.traitlets, pkgs.python37Packages.pandas, pkgs.python37Packages.datashape, pkgs.python37Packages.pandas, pkgs.python37Packages.datashape, pkgs.python27Packages.traitlets, pkgs.python37Packages.pandas, pkgs.python27Packages.datashape, pkgs.python27Packages.pandas, pkgs.python27Packages.datashape, pkgs.python37Packages.traitlets, pkgs.python27Packages.pandas, pkgs.python27Packages.datashape
- @zimbatm for pkgs.python37Packages.pip-tools, pkgs.python27Packages.pip-tools, pkgs.python37Packages.pip-tools, pkgs.python27Packages.pip-tools, pkgs.python37Packages.pip-tools, pkgs.python27Packages.pip-tools, pkgs.python37Packages.pip-tools, pkgs.python27Packages.pip-tools
- @7c6f434c for pkgs.python37Packages.pandas, pkgs.python27Packages.pandas, pkgs.python27Packages.pandas, pkgs.python37Packages.pandas, pkgs.python37Packages.pandas, pkgs.python37Packages.pandas, pkgs.python27Packages.pandas, pkgs.python27Packages.pandas
- @catern for pkgs.python37Packages.trio, pkgs.python37Packages.trio, pkgs.python37Packages.trio, pkgs.python37Packages.trio

does this seem reasonable? please discuss on https://github.com/NixOS/ofborg/pull/293

File: pkgs/development/python-modules/pytest/default.nix [hunk 1]
```diff
@@ -3,8 +3,35 @@
...
-  buildPythonPackage rec {
+let generic = { version, sha256 }: let
+
+  # runTests is a function that returns a string that can be used as a checkPhase.
+  runTests = with stdenv.lib; {
+    disabledTests ? [],     # Disable tests that match expression.
+    disabledTestIds ? [],   # Disable tests that match given node id's. This is useful in case a test in a specific file needs to be disabled.
+    options ? [],           # py.test options.
+    targets ? [],           # py.test files or folders.
+    variables ? {},         # Environment variables to export.
```
Comment (Mic92) [inline pkgs/development/python-modules/pytest/default.nix]: `variables` is a bit ambiguities. How about `environment` or `environmentVariables`?

**Comment by FRidh** [comment]
`runTests` returns a string, and can therefore unfortunately not be overridden. If, instead, `buildPython*` would accept a parameter, say, `checkConfig`, whose value corresponds to an attribute set, then it would be possible to write:
```
buildPythonPackage {
  ...

  checkConfig = {
    runner = pytest;
    variables.FOO = "bar";
    disabledTests = [
      "test_this"
    ];
  };
}
```
where the runner gets to decide how to invoke the test.
I don't want to over-engineer it (mainly just have a solution for `disabledTests`), but considering we have different test runners it may be convenient. At the same time, `nose` is being used fewer and fewer.

**Comment by Mic92** [comment]
Just two points here?

1. Was there a particular reason not to use a build-support hook?
2. Can we get documentation for this function in the manual?

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by FRidh** [comment]
Will continue with hook approach.

**Pull request closed without merging**